miden-lib/asm/note_scripts/basic.masm

```diff
@@ -0,0 +1,75 @@
+use.miden::sat::account
+use.miden::sat::note
+use.miden::sat::tx
+use.miden::wallets::basic->wallet
+
+#! Pay to ID - ensures that note can only be consumed by an account with the provided ID.
+#! 
+#! Inputs: [target_account_id]
+#! Outputs: []
+#!
+#! - ASSET is the asset to be received, can be fungible or non-fungible
+#!
+#! FAILS if: 
+#! - Account ID of executing account is not equal to specified Account ID
+#! - The same non-fungible asset already exists in the account.
+#! - Adding a fungible asset would result in amount overflow, i.e.,
+#!   the total amount would be greater than 2^63.
+export.p2id                                             # [note_inputs = target_account_id, ...]                                            
+    exec.account::get_id                                # [account_id, target_account_id, ...]
+    
+    assert_eq                                           # [] if account_id == target_account_id, fails if not
+
+    push.0                                              # [0, ...] memory pointer to store assets
+    exec.note::get_assets                               # [num_of_assets, 0, ...] 
+
+    dup neq.0                                           # [1 || 0, num_of_assets, 0, ...], sanity check that there is at least one asset
+    while.true                                          
+        sub.1 padw swap.5 dup swap.6                    # [num_of_assets - (i), 0, 0, 0, 0, num_of_assets - (i), ...], i ~ round in the loop
+        mem_loadw                                       # [ASSET, num_of_assets - (i), ...], i ~ round in the loop
+        call.wallet::receive_asset dropw                # [num_of_assets - (i), ...], i ~ round in the loop
+        dup neq.0                                       # [1 || 0, num_of_assets - (i), ...], if end of memory reached, break
+    end
+
+    drop drop
+end
+
+#! Pay to ID reclaimable - ensures that note can only be consumed by an account with the provided ID
+#! or after provided block number only by the sender account
+#! 
+#! Inputs: # [reclaim_block_height, target_account_id, sender_account_id, ...]
+#! Outputs: []
```

Comment:
`sender_account_id` would not be provided via note inputs. Instead, we can get it directly from the note via [note::get_sender](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-lib/asm/sat/note.masm#L59).

Reply:
changed it